packages/react-router/src/useHoistedRoutes.ts

```diff
@@ -0,0 +1,68 @@
+import type { RootRoute, Route } from "./routeRegistry.ts";
+import { useMemo, useState } from "react";
+
+import { isNil } from "@squide/core";
+
+export type WrapManagedRoutesFunction = (routes: Route[]) => Route;
```

Comment:
is it really worth it to name this type?

```
export interface UseHoistedRoutesOptions {
    wrapManagedRoutes?: (routes: Route[]) => Route;
    allowedPaths?: string[];
}
```

if we don't expect people to use it, i wouldn't name it. In the low chances that people need to type something with it, they can use 
```
const myTestVariable: UseHoistedRoutesOptions ["wrapManagedRoutes"];
```